### DISCUSSION TITLE
Set mime type when adding attachment to issue?

### DISCUSSION THREAD

**Discussion by GrantEdwards** [pycontribs/jira#1506] [Q&A]
When attaching a file to an issue, is there a way to set the mime type?

I'm migrating issues from MantisBT, and the attached files all have mime types associated with them so that they can be handled intelligently (both by the web UI and when being downloaded). Do Jira attachments have file type info?

**Comment by adehad** [comment]
@GrantEdwards 
Sorry for the delay, at the moment there is no specific facility that has been implemented with that in mind. My understanding is that it currently uses `application/octet-stream` but perhaps we can use https://docs.python.org/3/library/mimetypes.html to guess the type of a file.

If you do end up implementing a solution that you are happy to share we would happily consider a pull request !
